src/overlays/actors/ovl_En_Bigpamet/z_en_bigpamet.c

```diff
@@ -90,121 +113,757 @@ static DamageTable D_80A29734 = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A28378.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A283A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A283F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A2844C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A284E4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A2855C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A28618.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A2866C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A286C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A28708.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A28760.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_S8(hintId, 1, ICHAIN_CONTINUE),    ICHAIN_VEC3F_DIV1000(scale, 15, ICHAIN_CONTINUE),
+    ICHAIN_F32(gravity, -2, ICHAIN_CONTINUE), ICHAIN_F32(targetArrowOffset, 4333, ICHAIN_CONTINUE),
+    ICHAIN_U8(targetMode, 5, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A287E8.s")
+s32 D_80A29778 = 0;
+Vec3f D_80A2977C = { 0.0f, 1.0f, 0.0f };
+Color_RGBA8 D_80A29788 = { 250, 250, 250, 255 };
+Color_RGBA8 D_80A2978C = { 180, 180, 180, 255 };
+
+void EnBigpamet_Init(Actor* thisx, PlayState* play) {
+    EnBigpamet* this = THIS;
+    s32 i;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+
+    SkelAnime_InitFlex(play, &this->skelAnime2, &object_tl_Skel_007C70, &object_tl_Anim_004210, this->jointTable2,
+                       this->morphTable2, OBJECT_TL_2_LIMB_MAX);
+    SkelAnime_InitFlex(play, &this->skelAnime1, &object_tl_Skel_001A50, &object_tl_Anim_000B30, this->jointTable1,
+                       this->morphTable1, OBJECT_TL_1_LIMB_MAX);
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 55.0f);
+
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    if (!D_80A29778) {
+        for (i = 0; i < ARRAY_COUNT(D_80A29754); i++) {
+            D_80A29754[i] = Lib_SegmentedToVirtual(D_80A29754[i]);
+        }
+        D_80A29778 = true;
+    }
+
+    this->actor.params = ENBIGPAMET_1;
+    func_80A281B4(this);
+}
+
+void EnBigpamet_Destroy(Actor* thisx, PlayState* play) {
+    EnBigpamet* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void func_80A2768C(EnBigpamet* this) {
+    if (this->unk_29C != 0) {
+        this->unk_29C++;
+        if (this->unk_29C == 4) {
+            this->unk_29C = 0;
+        }
+    } else if (Rand_ZeroOne() < 0.05f) {
+        this->unk_29C = 1;
+    }
+}
+
+void func_80A276F4(EnBigpamet* this) {
+    if (this->actor.parent->params == 3) {
```

Comment:
```suggestion
    if (this->actor.parent->params == GEKKO_ON_SNAPPER) {
```
You'll want to include `z_en_pametfrog.h` for this